src/test/java/org/jenkinsci/plugins/workflow/libs/LibraryAdderTest.java

```diff
@@ -336,4 +337,22 @@
         r.assertLogContains("set to loaded null", r.buildAndAssertSuccess(p));
     }
 
+    @Ignore
+    @Issue("JENKINS-56682")
+    @Test public void scriptVarsWhereInitializerUsesLibrary() throws Exception {
+        sampleRepo.init();
+        sampleRepo.write("src/pkg/Foo.groovy", "package pkg; class Foo { }");
+        sampleRepo.git("add", "src");
+        sampleRepo.git("commit", "--message=init");
+        GlobalLibraries.get().setLibraries(Collections.singletonList(
+            new LibraryConfiguration("lib",
+                new SCMSourceRetriever(new GitSCMSource(null, sampleRepo.toString(), "", "*", "", true)))));
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "@Library('lib@master') import pkg.Foo\n" +
+                "import groovy.transform.Field\n" +
+                "@Field f = new Foo()\n", true));
```

Comment:
Also check if something like this works:

```groovy
static def f = new Foo()
```

Reply:
It looks like that works fine as-is.

Reply:
Just fishing around for ways to convince the Groovy compiler to create an `<init>` or `<clinit>` method in `WorkflowScript`. The Java syntax

```java
{
  // …
}
```

presumably does not work in Groovy. Not sure if you can declare an explicit constructor

```groovy
def WorkflowScript() {
  new Foo()
}
```

etc.

Reply:
Ok, it looks like if your Groovy script just declares a class, that gets used for the Script class, and you can use (static) initializer blocks as you would in Java, and they all throw similar errors as for `@Field` during `GroovyShell.parse`.

Error for a regular initializer block:
```
org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use new pkg.Foo
	at org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.StaticWhitelist.rejectNew(StaticWhitelist.java:271)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onNewInstance(SandboxInterceptor.java:174)
	at org.kohsuke.groovy.sandbox.impl.Checker$3.call(Checker.java:198)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedConstructor(Checker.java:203)
	at org.kohsuke.groovy.sandbox.impl.Checker$checkedConstructor.callStatic(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:194)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:214)
	at MyScript.<init>(WorkflowScript:3)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at org.codehaus.groovy.runtime.InvokerHelper.createScript(InvokerHelper.java:438)
Caused: groovy.lang.GroovyRuntimeException: Failed to create Script instance for class: class MyScript. Reason
	at org.codehaus.groovy.runtime.InvokerHelper.createScript(InvokerHelper.java:466)
	at groovy.lang.GroovyShell.parse(GroovyShell.java:700)
```

Error for a static initializer block:

```
org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use new pkg.Foo
	at org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.StaticWhitelist.rejectNew(StaticWhitelist.java:271)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onNewInstance(SandboxInterceptor.java:174)
	at org.kohsuke.groovy.sandbox.impl.Checker$3.call(Checker.java:198)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedConstructor(Checker.java:203)
	at org.kohsuke.groovy.sandbox.impl.Checker$checkedConstructor.callStatic(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:194)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:214)
	at MyScript.<clinit>(WorkflowScript:3)
Caused: java.lang.ExceptionInInitializerError
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at org.codehaus.groovy.runtime.InvokerHelper.createScript(InvokerHelper.java:438)
	at groovy.lang.GroovyShell.parse(GroovyShell.java:700)
```

Reply:
> `at MyScript.<init>(WorkflowScript:3)`

Hmm, a corner case. See [JENKINS-31838](https://issues.jenkins-ci.org/browse/JENKINS-31838).